### ISSUE TITLE
maybe do not consider the ignore label in the training data

### ISSUE THREAD

**Issue by ceo1207** [divamgupta/image-segmentation-keras#52]
the training you use is CamVid, which contains 11 classes [0,10] and 1 [11] ignored class, labeling the pixel from [0,10] and 11. When calculate the categorical cross entropy, you should ignore the loss from the pixel which is labeled 11. But I do not find the coresponding operation.

**Comment by ceo1207** [comment]
ok， I think I am wrong. You set the 11th label pixel  into a full of zero in one hot encoding. So when calculate the cross entropy, it will not influence the loss. Am I right?

**Comment by JordanMakesMaps** [comment]
@ceo1207, I believe you're right. I've spent some time trying to figure this out in Keras because nowhere I've seen has it be explicitly written. What I end up doing is creating a mask for the image, and anything within the mask I do not want seen by the image I label it as ignore or background, and denote those pixels as (0, 0, 0) or (0) depending on if you're creating a RGB mask or a grayscale. Then when it comes to feeding the data to the model during training, create a one-hot-encoded mask using `to_categorical` and then with the output, replace the entire ignore/background channel with zeros.

My issue was that I was creating background/ignore as it's own class which gave really bad results (the model would predict everything as background/ignore). Hope that helps someone in the future!

**Comment by hassanrazakhalid** [comment]
@JordanMakesMaps  can you please explain where to put ignore labels stuff. Here is the sample code from the site.. The sample code just takes the path for training images and labels along with few other params. Thanks
```
from keras_segmentation.models.unet import vgg_unet

model = vgg_unet(n_classes=51 ,  input_height=416, input_width=608  )

model.train(
    train_images =  "dataset1/images_prepped_train/",
    train_annotations = "dataset1/annotations_prepped_train/",
    checkpoints_path = "/tmp/vgg_unet_1" , epochs=5
)
```

**Comment by JordanMakesMaps** [comment]
@hassanrazakhalid I ended up using @ceo1207's idea and used a weighted loss function so the ignore class has a weight of 0, and the other classes that are of interest are equal to each other.

Here's the code from @qubvel's repo:

https://github.com/qubvel/segmentation_models/blob/master/segmentation_models/losses.py
